stacks_queues/queue.py

```diff
@@ -10,53 +10,110 @@ class QueueEmptyException(Exception):
 class Queue:
 
     def __init__(self):
-        self.store = [None] * INITIAL_QUEUE_SIZE
+        self.store = [None] * INITIAL_QUEUE_SIZE 
         self.buffer_size = INITIAL_QUEUE_SIZE
+        # pointers
         self.front = -1
         self.rear = -1
+        # How many elements are already in the cb - initially zero 
         self.size = 0
-      
-
+    
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        # Queue is full
+        if self.size >= self.buffer_size:  
+            raise QueueFullException("Queue is Full")
+
+        # if rear pointer is past the allowed size, cannot insert anymore, reset it 
+        if self.rear >= (self.buffer_size - 1):   # index, not size
+            #reset rear
+            self.rear = -1
+
+        # after checking queue is not full & there's space in queue we
+        # can insert it and need to move rear and update size to keep track 
+        self.store[self.rear + 1] = element 
+        # keep track of how many elements in array / where end of queue is
+        self.size += 1
+        self.rear += 1
 
     def dequeue(self):
         """ Removes an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
             returns None
         """
-        pass
```

Comment:
👍 These work, but it's simpler to use the % operator to calculate the next index.